lib/OpenQA/Utils.pm

```diff
@@ -164,6 +166,35 @@ sub testcasedir {
     return $dir;
 }
 
+sub gitrepodir {
+    my %args = (
+        distri  => '',
+        version => '',
+        repo    => 'tests',
+        @_,
+    );
+    my $path
+      = $args{repo} eq 'needles'
+      ? needledir($args{distri}, $args{version})
+      : testcasedir($args{distri}, $args{version});
```

Comment:
As this is the only use of the "repo" argument I suggest to make the argument a boolean, e.g. `needles => 1` instead of `repo => 'needles'`

Reply:
*this change is ready for review*